tests/xharness/Execution/ProcessManager.cs

```diff
@@ -7,84 +7,100 @@
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Xml;
 using Xharness.Execution.Mlaunch;
 using Xharness.Logging;
 using Xharness.Utilities;
 
 namespace Xharness.Execution {
 	public class ProcessManager : IProcessManager {
-		public ProcessManager ()
+		public string XcodeRoot { get; }
+		public string MlaunchPath { get; }
+
+		Version xcode_version;
+		public Version XcodeVersion {
+			get {
+				if (xcode_version == null) {
+					var doc = new XmlDocument ();
+					doc.Load (Path.Combine (XcodeRoot, "Contents", "version.plist"));
+					xcode_version = Version.Parse (doc.SelectSingleNode ("//key[text() = 'CFBundleShortVersionString']/following-sibling::string").InnerText);
+				}
+				return xcode_version;
+			}
+		}
+
+		public ProcessManager (string xcodeRoot, string mlaunchPath)
 		{
+			XcodeRoot = xcodeRoot ?? throw new ArgumentNullException (nameof (xcodeRoot));
+			MlaunchPath = mlaunchPath ?? throw new ArgumentNullException (nameof (mlaunchPath));
 		}
 
 		public async Task<ProcessExecutionResult> ExecuteCommandAsync (string filename,
 			IList<string> args,
 			ILog log,
 			TimeSpan timeout,
-			Dictionary<string, string> environment_variables = null,
-			CancellationToken? cancellation_token = null)
+			Dictionary<string, string> environmentVariables = null,
+			CancellationToken? cancellationToken = null)
 		{
-			using (var p = new Process ()) {
-				p.StartInfo.FileName = filename;
-				p.StartInfo.Arguments = StringUtils.FormatArguments (args);
-				return await RunAsync (p, log, timeout, environment_variables, cancellation_token);
-			}
+			using var p = new Process ();
+			p.StartInfo.FileName = filename ?? throw new ArgumentNullException (nameof (filename));
+			p.StartInfo.Arguments = StringUtils.FormatArguments (args);
+			return await RunAsync (p, log, timeout, environmentVariables, cancellationToken);
 		}
 
-		public async Task<ProcessExecutionResult> ExecuteCommandAsync (string filename,
-			MlaunchArguments args,
+		public async Task<ProcessExecutionResult> ExecuteCommandAsync (MlaunchArguments args,
 			ILog log,
 			TimeSpan timeout,
-			Dictionary<string, string> environment_variables = null,
-			CancellationToken? cancellation_token = null)
+			Dictionary<string, string> environmentVariables = null,
+			CancellationToken? cancellationToken = null)
 		{
-			using (var p = new Process ()) {
-				p.StartInfo.FileName = filename;
-				p.StartInfo.Arguments = args.AsCommandLine ();
-				return await RunAsync (p, log, timeout, environment_variables, cancellation_token);
-			}
+			return await RunAsync (new Process(), args, log, timeout, environmentVariables, cancellationToken);
```

Comment:
We are leaking the Process here! Or at least, it will not be disposed. Either re-add the using block, or move too the new using pattern.

Reply:
Good catch